lib/Models/SdmxJsonCatalogItem.js

```diff
@@ -529,39 +575,55 @@ function getTotalValueIdsForDimensionId(item, dimensionId) {
  *             - id
  *             - name
  * If there is a whitelist, only the whitelisted values are included.
+ * If there is a blacklist, blacklisted values are excluded.
  * @private
  * @param  {SdmxJsonCatalogItem} item The SDMX-JSON catalog item.
  * @param  {Array} structureSeries The structure's series property, json.structure.dimensions.series.
  * @return {Object[]} A description of the dimensions.
  */
 function buildDimensions(item, structureSeries) {
-    // Store the length of each dimension, in the correct keyPosition.
-    function getWhitelistFilter(dimensionId) {
-        var thisIdsWhitelist = item.whitelist[dimensionId];
-        if (!defined(thisIdsWhitelist)) {
+    // getFilter returns a function which can be used in list.filter().
+    // It tests if the value is in the given list, using regexps if possible.
+    // filterList can be either item.whitelist or item.blacklist.
+    // set isWhiteList false if is blacklist, so that the return values are negated (except for a missing list).
+    function getFilter(filterList, dimensionId, isWhiteList) {
+        var thisIdsFilterList = filterList[dimensionId];
+        if (!defined(thisIdsFilterList)) {
             return function() { return true; };
         }
         try {
-            var thisIdsWhitelistRegExps = thisIdsWhitelist.map(string => new RegExp('^' + string + '$'));
+            var thisIdsRegExps = thisIdsFilterList.map(string => new RegExp('^' + string + '$'));
         } catch (e) {
             // Cannot intepret as a regular expression.
             // Eg. "[" causes Uncaught SyntaxError: Invalid regular expression: /[/: Unterminated character class(…)),
             // So just test as a string.
             return function(value) {
-                return thisIdsWhitelist.indexOf(value.id) >= 0;
+                var isPresent = thisIdsFilterList.indexOf(value.id) >= 0;
+                if (isWhiteList) {
+                    return isPresent;
+                }
+                return !isPresent;
             };
         }
         return function(value) {
             // Test as a straight string, and if that fails, as a regular expression.
-            return thisIdsWhitelist.indexOf(value.id) >= 0 || thisIdsWhitelistRegExps.map(regExp => value.id.match(regExp)).some(defined);
+            var isPresent = thisIdsFilterList.indexOf(value.id) >= 0 || thisIdsRegExps.map(regExp => value.id.match(regExp)).some(defined);
```

Comment:
Slightly dodgy use of `thisIdsRegExps` here, because it's defined in a nested scope.  Of course, JavaScript doesn't pay attention to such things, but it's still a but strange.  I think moving the `return function(value) ...` inside the `try` as well would make it clearer.